### DISCUSSION TITLE
Tween when using set_limit_target

### DISCUSSION THREAD

**Discussion by jjolmo** [ramokz/phantom-camera#364] [Question / Help Wanted]
Hi!

I have multiple rooms, these rooms are designed by a limit defined by a CollisionShape2D. Could be possible that when you assign a new "limit_target" the camera will tween from where it is to the new position?

Think it as a cheap way to create a Zelda2D dungeon movement where the camera moves to one direction to position in the next room
For now I'm doing something quite complex which doesn't seem to work correctly but may need some tweak to make it possible:
1. Create a dummy point and position it on the middle of the room
2. Set the camera to follow this target. The camera moves with some cool tweeting animation
3. Once the animation has ended: Set the camera bound to the new room, just in case you want to have different sizes of dungeons

See this image that displays how a big room with a different size would be (marked with a cross)
![image](https://github.com/user-attachments/assets/712bb6da-78c1-4626-a731-3245a86928be)

I expect to change the limit  of the camera whenever the user enters a new room which are separated by area2D collisionShapes

thanks!

**Comment by ramokz** [comment]
For this, you would generally use a different PCam for each room and then tween between each as you enter and leave the different areas.

There was another discussion post about this, where you can see what that translated to: https://github.com/ramokz/phantom-camera/discussions/195#discussioncomment-8387648
